### PR TITLE
Slashing related e2e test improvements

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
Improves a couple slashing related e2e tests by:
1. Removing duplicate test/setup code in favor of test cases and if-statements (TestSendSlashPacketDowntime and TestSendSlashPacketDoubleSign were 90% the same, now they are one test method)
2. Removing a hack which previously reconstructed a VSC packet from local variables, and manually sent/received that packet using IBC testing functions
3. Using `providerSlashingKeeper.SlashFractionDowntime` instead of hardcoding an expected slash fraction

The second point here is needed to properly tie https://github.com/cosmos/interchain-security/pull/462 into existing e2e tests, the hack was faultily making e2e tests fail.

### CODE REVIEW

File: tests/e2e/slashing.go [hunk 1]
```diff
@@ -14,249 +14,170 @@ import (
...
-		Address: tmVal.Address,
-		Power:   tmVal.VotingPower,
-	}
+		// check that the validator was removed from the provider validator set
+		s.Require().Len(s.providerChain.Vals.Validators, validatorsPerChain-1)
 
-	oldBlockTime := s.consumerCtx().BlockTime()
-	packetData := types.NewSlashPacketData(validator, valsetUpdateId, stakingtypes.DoubleSign)
+		// Relay the VSC packet to the consumer
+		relayAllCommittedPackets(s, s.providerChain, s.path, ccv.ProviderPortID, s.path.EndpointB.ChannelID, 1)
```
Comment (smarshall-spitzbart) [inline tests/e2e/slashing.go:125]: This is the new way that we relay the VSC packet from provider to consumer. Note that this method is commonly used elsewhere

File: tests/e2e/slashing.go [hunk 2]
```diff
@@ -14,249 +14,170 @@ import (
...
+		// check that the validator's tokens were slashed
+		var slashFraction sdk.Dec
+		if tc == downtimeTestCase {
+			slashFraction = providerSlashingKeeper.SlashFractionDowntime(s.providerCtx())
 
-	// update consumer client on the VSC packet sent from provider
-	err = s.path.EndpointA.UpdateClient()
-	s.Require().NoError(err)
-
-	// reconstruct VSC packet
```
Comment (smarshall-spitzbart) [inline tests/e2e/slashing.go:214]: This was the old way we sent the VSC packet from provider to consumer

File: x/ccv/provider/keeper/keeper.go [hunk 3]
```diff
@@ -830,6 +830,24 @@ func (k Keeper) SetVscSendTimestamp(
 	store.Set(types.VscSendingTimestampKey(chainID, vscID), timeBz)
 }
 
+// GetVscSendTimestamp returns a VSC send timestamp by chainID and vscID
+//
+// Note: This method is used only for testing.
```
Comment (danwt) [inline x/ccv/provider/keeper/keeper.go:835]: 👍

**Review by danwt — APPROVED** [review]
Really nice work, the new test is very clear 👍

**Review by sainoe — APPROVED** [review]
LGTM! Just a small comment.

File: tests/e2e/slashing.go (hunk 1 above)
Comment (smarshall-spitzbart) [reply tests/e2e/slashing.go:132]: I've also updated the comment for `TestRelayAndApplySlashPacket` to make this more clear

**Pull request merged**